src/services/__tests__/MatchService.unit.spec.ts

```diff
@@ -13,33 +14,64 @@ beforeEach(() => {
 });
 
 describe("Match Service tests", () => {
-  it("random teams are created", async () => {
-    const mockBallChasers = BallChaserQueueBuilder.many(6, { team: null });
-    jest.mocked(QueueRepository.getAllBallChasersInQueue).mockResolvedValueOnce(mockBallChasers);
+  describe("Random button was pressed", () => {
+    it("random teams are created", async () => {
+      const mockBallChasers = BallChaserQueueBuilder.many(6, { team: null });
+      jest.mocked(QueueRepository.getAllBallChasersInQueue).mockResolvedValueOnce(mockBallChasers);
 
-    const players = await createRandomMatch();
+      const players = await createRandomMatch();
 
-    expect(players.filter((player) => player.team === Team.Orange)).toHaveLength(3);
-    expect(players.filter((player) => player.team === Team.Blue)).toHaveLength(3);
-  });
+      expect(players.filter((player) => player.team === Team.Orange)).toHaveLength(3);
+      expect(players.filter((player) => player.team === Team.Blue)).toHaveLength(3);
+    });
+    it("updates queue with update player teams", async () => {
+      const mockBallChasers = BallChaserQueueBuilder.many(6, { team: null });
+      jest.mocked(QueueRepository.getAllBallChasersInQueue).mockResolvedValueOnce(mockBallChasers);
+
+      await createRandomMatch();
+      const addMatchMock = jest.mocked(ActiveMatchRepository.addActiveMatch);
 
-  it("updates queue with update player teams", async () => {
-    const mockBallChasers = BallChaserQueueBuilder.many(6, { team: null });
-    jest.mocked(QueueRepository.getAllBallChasersInQueue).mockResolvedValueOnce(mockBallChasers);
+      expect(addMatchMock).toHaveBeenCalled();
+    });
+    it("removes players from queue", async () => {
+      const mockBallChasers = BallChaserQueueBuilder.many(6);
+      jest.mocked(QueueRepository.getAllBallChasersInQueue).mockResolvedValueOnce(mockBallChasers);
 
-    await createRandomMatch();
-    const addMatchMock = jest.mocked(ActiveMatchRepository.addActiveMatch);
+      await createRandomMatch();
+      const removePlayersMock = jest.mocked(QueueRepository.removeAllBallChasersFromQueue);
 
-    expect(addMatchMock).toHaveBeenCalled();
+      expect(removePlayersMock).toHaveBeenCalled();
+    });
   });
+  describe("Captains button was pressed", () => {
+    it("last available queueMember is placed on blue team", async () => {
+      const mockBallChasers: PlayerInQueue[] = [];
+      const mockOrangeBallChasers = BallChaserQueueBuilder.many(3, { team: Team.Orange });
+      const mockBlueBallChasers = BallChaserQueueBuilder.many(2, { team: Team.Blue });
+      const mockBallChaserNoTeam = BallChaserQueueBuilder.single({ team: null, isCap: false });
+      mockBallChasers.push(mockBallChaserNoTeam);
+      mockOrangeBallChasers.forEach((p) => {
+        mockBallChasers.push(p);
+      });
+      mockBlueBallChasers.forEach((p) => {
+        mockBallChasers.push(p);
+      });
```

Comment:
This can be simplified to:
```
const mockBallChasers = [
   ...mockOrangeBallChasers,
   ...mockBlueBallChasers,
   mockBallChaserNoTeam,
];
```

Reply:
Fixed in next commit.